### PR TITLE
Sheets API is integrated

### PR DESCRIPTION
**Pull request by artemstepanenko**

### CODE REVIEW

**Comment by milenvoutchev** [comment]
Please add the relevant `config.[dev|prod].js` with the required structure, but with dummy values. This way the developer or DevOps guys just needs to change the values, not create new files and figure out what's the required structure inside.

File: server/src/config/config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* eslint import/no-dynamic-require: 0 */
+const config = require(`./config.${process.env.NODE_ENV}`);
+
+export const spreadsheetID = () => (config.default.spreadsheetID);
```
Comment (milenvoutchev) [inline server/src/config/config.js]: Please group the related config items together, e.g. under `bugsHistory` so that we know in future why is it there.

Comment (milenvoutchev) [inline server/src/config/config.js]: Why the need to export each config variable separately and as function? How about just importing `config` where they are used, then use as `config.spreadsheetID`

File: server/src/routes/bugsHistory.js [hunk 2]
```diff
@@ -1,36 +1,52 @@
...
-          bugs: 9,
-        },
-        {
-          label: 'Thu',
-          bugs: 10,
-        },
-        {
-          label: 'Fri',
-          bugs: 7,
+const jwtClient = new google.auth.JWT(
```
Comment (milenvoutchev) [inline server/src/routes/bugsHistory.js]: Please see if we can abstract the connection and `get`, `try-catch` logic. We might use Google Sheets as source for another widget.

File: server/package.json [hunk 3]
```diff
@@ -3,18 +3,21 @@
   "version": "0.0.0",
   "private": true,
   "scripts": {
-    "start:dev": "nodemon -w src --exec \"npm run lint:js && babel-node src --presets env\"",
-    "start:build": "node build",
+    "start:dev": "nodemon -w src --exec \"export NODE_ENV=dev&&npm run lint:js && babel-node src --presets env\"",
```
Comment (gkatsanos) [inline server/package.json]: should we space out && and the left/right parts?

File: server/src/config/config.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/* eslint import/no-dynamic-require: 0 */
+import _ from 'lodash';
```
Comment (gkatsanos) [inline server/src/config/config.js]: you don't need to import the entire lodash library, they have modularized most of their functions : https://www.npmjs.com/package/lodash.defaultsdeep ;)

File: server/package.json [hunk 5]
```diff
@@ -3,18 +3,21 @@
...
   },
   "dependencies": {
+    "ajv": "^5.1.5",
     "body-parser": "~1.17.1",
     "chalk": "^1.1.3",
     "cookie-parser": "~1.4.3",
     "express": "^4.15.2",
+    "googleapis": "^19.0.0",
     "jade": "~1.11.0",
+    "lodash": "^4.17.4",
```
Comment (gkatsanos) [inline server/package.json]: replace with https://www.npmjs.com/package/lodash.defaultsdeep

File: server/src/fetchSheet.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    fields: 'sheets(data(rowData(values(userEnteredValue))))',
+  };
+  sheets.spreadsheets.get(request, (err, response) => {
+    if (err) {
+      callback(false);
+      return;
+    }
+    try {
+      // @TODO Decouple "fetching" and "data crunching" so we can reuse them
+      const data = _.get(response, 'sheets[0].data');
```
Comment (gkatsanos) [inline server/src/fetchSheet.js]: https://www.npmjs.com/package/lodash.get but I'm wondering if we can do this natively? with default values from ES6?
http://exploringjs.com/es6/ch_parameter-handling.html#sec_parameter-default-values

File: server/package.json (hunk 3 above)
Comment (milenvoutchev) [inline server/package.json]: yes it would be more visible

File: server/package.json (hunk 5 above)
Comment (milenvoutchev) [inline server/package.json]: no need - added change to only import the used method

**Pull request merged**